--- a/src/corelib/global/qsysinfo.cpp
+++ b/src/corelib/global/qsysinfo.cpp
@@ -28,7 +28,7 @@
 
 #ifdef Q_OS_DARWIN
 #  include <private/qcore_mac_p.h>
-#  if __has_include(<IOKit/IOKitLib.h>)
+#  if defined(Q_OS_MACOS) && __has_include(<IOKit/IOKitLib.h>)
 #    include <IOKit/IOKitLib.h>
 #  endif
 #endif
@@ -1027,7 +1027,7 @@
 */
 QByteArray QSysInfo::machineUniqueId()
 {
-#if defined(Q_OS_DARWIN) && __has_include(<IOKit/IOKitLib.h>)
+#if defined(Q_OS_MACOS) && __has_include(<IOKit/IOKitLib.h>)
     char uuid[UuidStringLen + 1];
     io_service_t service = IOServiceGetMatchingService(kIOMainPortDefault, IOServiceMatching("IOPlatformExpertDevice"));
     QCFString stringRef = (CFStringRef)IORegistryEntryCreateCFProperty(service, CFSTR(kIOPlatformUUIDKey), kCFAllocatorDefault, 0);
